Update NFTS Pallet Information (#4452)
w3f/polkadot-wiki: docs/learn/learn-nft-pallets.md

## NFTs Pallet

NFTs is a [FRAME pallet](https://polkadot.js.org/docs/substrate/extrinsics#nfts) currently deployed
on Westmint, which provides a multitude of functions to interact with NFTs.

The pallet comes with a new way to configure NFTs, as well as configure collections and items.
Pallet-level [feature flags](https://github.com/paritytech/substrate/pull/12367) allow disabling
functionalities that are not needed in the runtime.

### Creating a Collection

:::info

Walkthrough video tutorial about creating a collection.

:::

You can use the NFTs pallet to create NFTs collections. In the Polkadot-JS UI go to Developer >
Extrinsic and select the `nfts.create` extrinsic. When you create a collection you must specify who
is the admin. Then, under `config: PalletNftsCollectionConfig`, you can configure your collection's
specifying different settings:

- `settings` you can specify (in a bitflag-format) settings for your collection:

>   - Transferrable items: items can be non-transferrable (good for soul-bound NFTs),
>   - Unlocked metadata: metadata can be locked,
>   - Unlocked attributes: attributes can be locked,
>   - Unlocked max supply: allows to change the max supply until it gets locked (i.e. possibility to
>     change the supply for a limited amount of times),
>   - Deposit required.

Reviewer: ```suggestion
  - `Transferrable items`: when disabled, the items will be non-transferrable (good for soul-bound NFTs),
  - `Unlocked metadata`: when disabled, the metadata will be locked,
  - `Unlocked attributes`: when disabled, the attributes in the `CollectionOwner` namespace will be locked,
  - `Unlocked max supply`: allows to change the max supply until it gets locked (i.e. the possibility to
    change the supply for a limited amount of time),
  - `Deposit required`: when disabled, no mint deposit will be taken for items. This option can be set by a super-user only.
```

> Everything is unlocked by default (bitflag `0`).

Reviewer: ```suggestion
Everything is unlocked by default (bitflag value `0`).
```

> - `maxSupply: Option` (toggle option) gives you the possibility to specify the maximum number of

Reviewer: ```suggestion
- `maxSupply` (toggle option) gives you the possibility to specify the maximum number of
```

## NFTs Pallet

NFTs is a [FRAME pallet](https://polkadot.js.org/docs/substrate/extrinsics#nfts) currently deployed
on Westmint, which provides a multitude of functions to interact with NFTs.

The pallet comes with a new way to configure NFTs, as well as configure collections and items.
Pallet-level [feature flags](https://github.com/paritytech/substrate/pull/12367) allow disabling
functionalities that are not needed in the runtime.

### Creating a Collection

> You can use the NFTs pallet to create NFTs collections. In the Polkadot-JS UI, go to Developer >
> Extrinsic and select the `nfts.create` extrinsic. When you create a collection, you must specify who
> the admin is. Then, under `config: PalletNftsCollectionConfig`, you can configure your collection's
> specifying different settings:

Reviewer 2: ```suggestion
You can use the NFTs pallet to create NFT collections. In the Polkadot-JS UI, go to Developer >
Extrinsic and select the `nfts.create` extrinsic. When you create a collection, you must specify who
the admin is. Then, under `config: PalletNftsCollectionConfig`, you can configure your collection by 
specifying different settings:
```

> - `settings` you can specify (in a bitflag-format) settings for your collection:

Reviewer 2: I see that I can only enter a positive number `settings: u64` here. Let's say I would like to disable `Transferrable items` - what would be the numeric value for `settings: u64` ? As none of the settings below are visible on the UI, an example would make sense (on how to set the numeric value for settings).

Author: Polkadot JS does not support bitflags. Settings are possible but at the moment only the default "0" is avaialble.

Reviewer 2: Got it! Let us add this as an info note here.

to lock. Also, unauthorized and/or unprivileged transfers can be prevented by locking mechanism
(unprivileged actions can be re-allowed anytime).

:::

With all these options, one can decide to modify the price of the collection's items and who can
mint, receive or buy items in that collection. Time constraints are available with `startBlock` and
`endBlock` parameters. It is thus possible, for example, to create a schedule in which holders of
items in collection A (`HolderOf` parameter) will be able to claim for free a limited number of NFTs
from Collection X (`maxSupply` parameter) only within a specific time frame. You can modify the
parameters so anyone can buy more NFTs from Collection X.

This can be useful for events such as Hackathons where participants who bought a ticket receive the
NFT ticket from Collection A. Then, all holders of at least one item in Collection A (i.e. all
ticket holders) will be given free avatar NFT from Collection X within the event schedule. After the
event, any additional remaining items in Collection X can be made available to the public through a
marketplace.

> The requirement to get the free avatar is being a holder of at least one NFT in Collection A. One
> can only claim the avatar specifying which NFT (i.e. the ID) owns in Collection A. The same NFT
> cannot be used twice, meaning that holders of multiple NFTs in Collection A (for example,
> participants to multiple Hackathons) can claim multiple avatars specific to each event.

Reviewer 2: I made a few corrections. It is mentioned that same NFT cannot be used twice. But if someone has multiple NFTs, they should be able to claim multiple avatars right?

```suggestion
The requirement to get the free avatar is to hold at least one NFT in Collection A. One
can only claim the avatar specifying which NFT (i.e. the ID) they own in Collection A. The same NFT
cannot be used twice. Holders of multiple NFTs in Collection A (for example,
participants in multiple Hackathons) can claim multiple avatars specific to each event.
```

:::warning Time frame must be updated

Someone trying to mint an NFT outside the specified time frame will trigger a `NoConfig` error, as
the collection’s admin has specified no configuration after the time frame ends. The collection's
admin must call the `updateMintSettings` extrinsic and add a new schedule or disable the block
number option.

:::

After you minted an NFT, check which NFT IDs you own under which collection. In the Polkadot-JS UI
go to Developer > Chain State > Storage, select the `nfts.account` extrinsic, and specify the
account owning the NFT and the collection ID. You can also see all your collections by selecting the
`collectionAccount` extrinsic.

When a collection is destroyed, no one can pick up the collection ID again (including the owner).

> When a new collection is created, a new ID will be generated and assigned to that collection.

Reviewer 2: Reordering chronological events

```suggestion
When a new collection is created, a new ID will be generated and assigned to that collection.
When a collection is destroyed, no one can pick up the collection ID again (including the owner).

```

### Minting an NFT

> You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the followings:

Reviewer 2: ```suggestion
You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the following:
```

  - `Deposit required`: when disabled, no mint deposit will be taken for items. This option can be
    set by a super-user only.

:::info

> Note that currently Polkadot-JS does not support bitflags. Leave the settings field as it is.
> Everything is unlocked by default (bitflag value `0`).

Reviewer 2: ```suggestion
Note that currently, Polkadot-JS UI does not support bitflags. Leave the settings field as it is.
Everything is unlocked by default (bitflag value `0`).
```

be used twice, meaning that holders of multiple NFTs in Collection A (for example participants to
multiple Hackathons) can claim multiple avatars specific to each event.

:::warning Time frame must be updated

Someone trying to mint an NFT outside the specified time frame will trigger a `NoConfig` error, as
no configuration has been specified by the collection's admin after the time frame ends. The
collection's admin must call the `updateMintSettings` extrinsic and add a new schedule or disable
the block number option.

:::

After you minted an NFT, it is possible to check which NFT IDs you own under which collection. In
the Polkadot-JS UI go to Developer > Chain State > Storage, select the `nfts.account` extrinsic,
specify the account owning the NFT and the collection ID. You can also see all the collections you
own by selecting the `collectionAccount` extrinsic.

### Minting an NFT

:::info

Walkthrough video tutorial about minting an NFT.

:::

You can mint an NFT using the `nfts.mint` extrinsic. You must then specify the followings:

- `collection`, the collection ID where you want to mint

> - `item`, the item ID (auto-incremented)

Reviewer: technically, the item ID is not auto-incremented
